src/v/cloud_storage/recursive_directory_walker.cc

```diff
@@ -51,11 +51,14 @@ recursive_directory_walker::walk(
         try {
             ss::file target_dir = co_await open_directory(target);
             auto sub = target_dir.list_directory(
-              [&files,
-               &current_cache_size,
-               &dirlist,
+              [_files = &files,
+               _current_cache_size = &current_cache_size,
+               _dirlist = &dirlist,
                _target{target},
                _tracker{tracker}](ss::directory_entry entry) -> ss::future<> {
+                  auto& files = *_files;
+                  auto& current_cache_size = *_current_cache_size;
+                  auto& dirlist = *_dirlist;
```

Comment:
Ahh @Lazin i didn't realize that this PR was here when I sent you a message about a previous PR not having review feedback applied. Thanks for following up!

BTW, I'm not sure you need to capture a pointer. I think you can still capture the reference and then create another reference on the stack. But either way.